test/parallel/test-util-styletext.js

```diff
@@ -46,6 +46,39 @@ assert.strictEqual(
   '\u001b[1m\u001b[31mtest\u001b[39m\u001b[22m',
 );
 
+assert.strictEqual(
+  util.styleText('red',
+                 'A' + util.styleText('blue', 'B', { validateStream: false }) + 'C',
+                 { validateStream: false }),
+  '\u001b[31mA\u001b[34mB\u001b[31mC\u001b[39m'
+);
+
+assert.strictEqual(
+  util.styleText('red',
+                 'A' + util.styleText(['bgRed', 'blue'], 'B', { validateStream: false }) +
+    'C', { validateStream: false }),
+  '\x1B[31mA\x1B[41m\x1B[34mB\x1B[31m\x1B[49mC\x1B[39m'
+);
+
+assert.strictEqual(
+  util.styleText('dim',
+                 'dim' +
+    util.styleText('bold', 'bold', { validateStream: false }) +
+  'dim', { validateStream: false }),
+  '\x1B[2mdim\x1B[1mbold\x1B[2mdim\x1B[22m'
```

Comment:
We're so close 😅

The last `dim` here will be bold _and_ dim right now

But should be just dim

i.e it should actually be `x1B[22mx1b[2mdim` (close-open) 

I'm not sure how you can do this without having a special case in the reducer for bold and dim though 🤔

Basically everything works as expected except this one case now I think

Just to help you get your head around it, it's like changing the if to do this: 

```ts
if (code[0] === bold || code[0] === dim) {
  return close + open; 
}
return open; 
```

Reply:
not sure if this is good, since people could change bold and dim